www/perl-lib/SMT/RegistrationSharing.pm

```diff
@@ -243,42 +218,41 @@ sub shareRegistration
             . '</tableData>';
     }
     $regXML .= '</registrationData>';
+    $dbh->disconnect();
+    _sendRegData($regXML, $log);
+}
 
-    # Send the registration data to the configured SMT siblings
-    if ( $regXML =~ /GUID/) {
-        my @smtSiblings = split /,/, $shareRegDataTargets;
-        for my $smtServer (@smtSiblings) {
-            my $url = "https://$smtServer/center/regsvc"
-                . '?command=shareregistration'
-                . '&lang=en-US&version=1.0';
-            my $response = _sendData($url, $regXML, $log);
-            if (! $response ) {
-                return;
-            }
-            if (! $response->is_success ) {
-                my $msg = $response->message;
-                my $details = $response->content;
-                my $guidMsg = "Could not share registration for $regGUID";
-                my $responseMsg = "Response: $msg";
-                my $detailsMsg = "Response: $details";
-                if ($log) {
-                    print $log $guidMsg . "\n";
-                    print $log $responseMsg . "\n";
-                    print $log $detailsMsg . "\n";
-                } else {
-                    $apache->warn($guidMsg);
-                    $apache->warn($responseMsg);
-                    $apache->warn($detailsMsg);
-                    _logShareRecord($smtServer,$url,$regXML);
-                }
-
-            } else {
-                _sharePreviousRegistrations($smtServer, $log);
-            }
-        }
+#
+# Share a single product registration. In the SCC paradigm products/modules
+# are registered one at a time and we do not know when the client considers
+# registration to be complete. Thus we share every product as a registration
+#
+sub shareProductRegistration
+{
+    my $regGUID = shift;
+    my $prodID = shift;
+    my $apache = Apache2::ServerUtil->server;
+    my $dbh = SMT::Utils::db_connect();
+    my $regXML = '<?xml version="1.0" encoding="UTF-8"?>'
+        . '<registrationData>';
+    # Get the registration data
+    my $statement = sprintf("SELECT * from Registration where GUID=%s and PRODUCTID=%s",
+                            $dbh->quote($regGUID),
+                            $dbh->quote($prodID));
```

Comment:
This could make problems. PRODUCTID is referencing Products.ID which is an autoincrement value.
It is guaranteed that this ID is the same on all SMT server. The PRODUCTDATAID column contains the ID from SCC which is the same on all SMT servers.
But this would make the code a bit more complex of cause.